assets/scss/_functions.scss

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------*\
+    Functions
+\*---------------------------------------------------------------------------*/
+// STR REPLACE FUNCTION
+// https://css-tricks.com/snippets/sass/str-replace-function/
+@function str-replace($string, $search, $replace: '') {
+  $index: str-index($string, $search);
+  @if $index {
+    @return str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace);
+  }
+  @return $string;
+}
+// URL ENCODER FOR INLINE SVGS
+// Encodes dangerous characters for maximum compatability (down to IE9)
+@function url-encode($string) {
+  $map: (
+    "%": "%25",
+    "<": "%3C",
+    ">": "%3E",
+    "!": "%21",
+    "*": "%2A",
+    "'": "%27",
+    '"': "%22",
+    "(": "%28",
+    ")": "%29",
+    "{": "%7B",
+    "}": "%7D",
+    "[": "%5B",
+    "]": "%5D",
+    ";": "%3B",
+    ":": "%3A",
+    "@": "%40",
+    "&": "%26",
+    "=": "%3D",
+    "+": "%2B",
+    "$": "%24",
+    ",": "%2C",
+    "/": "%2F",
+    "?": "%3F",
+    "#": "%23"
+  );
+  @each $search, $replace in $map {
+    $string: str-replace($string, $search, $replace);
+  }
+  @return url('data:image/svg+xml;charset=utf-8,' + $string);
+}
+
+// MEDIA QUERY FUNCTION
+// To write named media queries
+@mixin media-query($media-query) {
```

Comment:
IMO this is an abstraction for the sake of it. What benefit does it bring to use a string as a breakpoint name? I rather see exactly what size the query is targeting without having to reference to a file

Reply:
Essentially what you're asking is; "what is the benefit of using variables?"
The benefit is to ensure that the breakpoint is always the same in every place the variable is used.
For example, somebody could accidentally hit backspace when editing a media query to make:
(max-width: 80px)
instead of
(max-width: 800px)
which is both valid CSS, so would not produce an error, but would yield completely different design results.
Using a variable ensures this kind of mistake cannot happen, because misspelling the variable would produce an error.

However I can understand your point that perhaps "mobile" and "desktop" are not very meaningful names for breakpoints, so I can rename the variables to be like "above-800" and "below-800" so an editor can know from a glance which screen width the media query is targeting.